autogluon/utils/tabular/ml/models/tabular_nn/tabular_nn_model.py

```diff
@@ -577,23 +568,27 @@ def _get_types_of_features(self, df, skew_threshold, embed_min_categories, use_n
             Warning("Attempting to _get_types_of_features for TabularNeuralNetModel, but previously already did this.")
 
         feature_types = self.feature_metadata.get_type_group_map_raw()
-
         categorical_featnames = feature_types[R_CATEGORY] + feature_types[R_OBJECT] + feature_types['bool']
         continuous_featnames = feature_types[R_FLOAT] + feature_types[R_INT]  # + self.__get_feature_type_if_present('datetime')
         language_featnames = [] # TODO: not implemented. This should fetch text features present in the data
         valid_features = categorical_featnames + continuous_featnames + language_featnames
-        if len(categorical_featnames) + len(continuous_featnames) + len(language_featnames) != df.shape[1]:
+        if len(valid_features) < df.shape[1]:
             unknown_features = [feature for feature in df.columns if feature not in valid_features]
-            # print('unknown features:', unknown_features)
+            logger.log(15, f"TabularNeuralNetModel will additionally ignore the following columns: {unknown_features}")
             df = df.drop(columns=unknown_features)
             self.features = list(df.columns)
-            # raise ValueError("unknown feature types present in DataFrame")
+
+        self.features_to_drop = df.columns[df.isna().all()].tolist()  # drop entirely NA columns which may arise after train/val split
```

Comment:
so features to drop are in self.features? Is this intended? Why not just remove from self.features?

Reply:
Didn't do that because this comment made it unclear whether `self.features` can vary between bagged copies of model or not:

https://github.com/awslabs/autogluon/blob/master/autogluon/utils/tabular/ml/models/abstract/abstract_model.py#L191-L193

Reply:
Ok, we can cleanup before 0.1, the preprocess logic needs to be abstracted towards FeatureGenerator anyways.